src/Pagination/usePagination.test.ts

```diff
@@ -0,0 +1,107 @@
+import { renderHook } from "@testing-library/react";
+import { usePagination } from "./usePagination";
+
+describe("usePagination", () => {
+  describe("visible pages", () => {
+    it("shows all pages when they fit in window", () => {
+      const { result } = renderHook(() => usePagination({ totalPages: 5 }));
+      expect(result.current.visiblePages).toEqual([1, 2, 3, 4, 5]);
+    });
+
+    it("centers selected page when possible", () => {
+      const { result } = renderHook(() =>
+        usePagination({ totalPages: 10, selectedPageNumber: 6 }),
+      );
+      expect(result.current.visiblePages).toEqual([5, 6, 7]);
```

Comment:
The default `windowSize` is `5`, but we adjust it according to how many ellipsis are showing. With 6 selected out of 10 pages, there will be ellipsis on both sides, resulting in 3 visible pages (5 - 2 ellipsis) with the `6` centered.